zerver/lib/events.py

```diff
@@ -139,9 +143,9 @@ def fetch_initial_state_data(user_profile: UserProfile,
         # these manual entries are for those realm settings that don't
         # fit into that framework.
         state['realm_authentication_methods'] = realm.authentication_methods_dict()
-        state['realm_allow_message_editing'] = realm.allow_message_editing
-        state['realm_allow_community_topic_editing'] = realm.allow_community_topic_editing
-        state['realm_allow_message_deleting'] = realm.allow_message_deleting
+        state['realm_allow_message_editing'] = False if user_profile is None else realm.allow_message_editing
+        state['realm_allow_community_topic_editing'] = False if user_profile is None else realm.allow_community_topic_editing
+        state['realm_allow_message_deleting'] = False if user_profile is None else realm.allow_message_deleting
```

Comment:
These should probably be commented noting that we pretend these features are disabled because guests can't access them.

In general this commit should be split into ~10 little commits for details like this block that can be merged incrementally, since I'm sure there will be parts of this that are hard to integrate, and I'd like to merge the rest if we can.